### PR TITLE
Add checkstyle rule to catch log.info(...%s...)

### LINKED ISSUE #643: Add checkstyle for logging placeholders
https://github.com/palantir/gradle-baseline/blob/develop/gradle-baseline-java-config/resources/checkstyle/checkstyle.xml#L282

### PR DESCRIPTION
**Pull request by gsheasby**
No release notes or docs changes needed.

Fixes #643.

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/704)

### CODE REVIEW

File: config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -39,6 +39,11 @@
           <property name="option" value="top"/>
           <property name="ordered" value="true"/>
         </module>
+        <module name="RegexpSinglelineJava">
+            <property name="format" value="(?i)log(ger)?\.(debug|info|warn|error)\(.*%[sd]"/>
```
Comment (schlosna) [inline config/checkstyle/checkstyle.xml]: do you want to catch `trace` as well?

Comment (tpetracca) [inline config/checkstyle/checkstyle.xml]: should also check trace

Comment (tpetracca) [inline config/checkstyle/checkstyle.xml]: also probably not important but if the regex can no match on things with String.format that would be good, or maybe an entirely different check is using log. with string.format and we say to just use native log. capabilities with {}

Comment (joelea) [inline config/checkstyle/checkstyle.xml]: added the trace.

**Comment by tpetracca** [comment]
+1

**Pull request merged**